### ISSUE TITLE
Set 5

### ISSUE THREAD

**Issue by shehanazHaque** [shehanazHaque/AssignmentDenvey#2]
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import r2_score, mean_squared_error
file_path = '/mnt/data/housing.csv'

try:
    data = pd.read_csv(file_path)
    print("Dataset loaded successfully!")
    print("First 5 rows:\n", data.head())
    print("\nData Info:\n")
    print(data.info())
    print("\nMissing values:\n", data.isnull().sum())
except Exception as e:
    print("Error loading dataset:", e)

data.dropna(inplace=True)

# Select features and target
# Assuming 'median_house_value' is the target variable
# and excluding any non-numeric or non-feature columns like 'ocean_proximity' if present
if 'median_house_value' in data.columns:
    X = data.select_dtypes(include=[np.number]).drop('median_house_value', axis=1)
    y = data['median_house_value']
else:
    raise ValueError("Target column 'median_house_value' not found in dataset.")

#  (80% training, 20% testing)
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Train the Linear Regression model
model = LinearRegression()
model.fit(X_train, y_train)

# Make predictions
y_pred = model.predict(X_test)

# Evaluate model performance
r2 = r2_score(y_test, y_pred)
mse = mean_squared_error(y_test, y_pred)

print("\nModel Evaluation:")
print(f"R-squared score: {r2:.4f}")
print(f"Mean Squared Error: {mse:.2f}")
